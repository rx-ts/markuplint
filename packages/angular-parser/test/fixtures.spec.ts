import path from 'node:path'
import { promisify } from 'node:util'

import glob from 'glob'
import { mlTestFile } from 'markuplint'

const asyncGlob = promisify(glob)

test('fixtures', async () => {
  const files = await asyncGlob(path.resolve(__dirname, 'fixtures/*.html'))
  for (const filePath of files) {
    const { violations } = await mlTestFile(filePath)
    expect(violations).toMatchSnapshot(path.basename(filePath))
  }
})
