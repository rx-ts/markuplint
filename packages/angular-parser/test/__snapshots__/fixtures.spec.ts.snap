// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures: attr-duplication.html 1`] = `
Array [
  Object {
    "col": 19,
    "line": 1,
    "message": "Duplicate attribute name",
    "raw": "[type]",
    "ruleId": "attr-duplication",
    "severity": "error",
  },
  Object {
    "col": 39,
    "line": 1,
    "message": "Duplicate attribute name",
    "raw": "[attr.type]",
    "ruleId": "attr-duplication",
    "severity": "error",
  },
]
`;

exports[`fixtures: deprecated-element.html 1`] = `
Array [
  Object {
    "col": 1,
    "line": 1,
    "message": "Element is deprecated",
    "raw": "<font color=\\"red\\">",
    "ruleId": "deprecated-element",
    "severity": "error",
  },
]
`;
